ql/src/java/org/apache/hadoop/hive/ql/exec/tez/TezSessionPool.java

```diff
@@ -131,13 +131,14 @@ SessionType getSession() throws Exception {
       poolLock.lock();
       try {
         while ((result = pool.poll()) == null) {
-          notEmpty.await(100, TimeUnit.MILLISECONDS);
+          LOG.info("Awaiting Tez session to become available in session pool");
+          notEmpty.await(10, TimeUnit.SECONDS);
```

Comment:
You are changing the waiting time to 10s from 100s. If this is intentional, then it shouldn't be put into a commit where the commit message says that it's about improving logging.

Reply:
Hello @miklosgergely.  The current change wait for 100ms whereas I have changed this to wait until 10s.  I'm not sure what the value is of setting a timeout, it will always just loop again.  Since I do not see any value here, but I wouldn't want to remove this as part of this ticket, I simply increased it.  Logging once every 100ms would be far too verbose. So you see, this change is logging-related. :)

Reply:
Any chance this change can increase the time we actually wait to get a session from the pool assuming there is none currently available?
It looks to me that if the next session becomes available in the next 10ms with the new change we might wait 10s instead -- am I missing something ?

Reply:
@pgaref This is not just a loop-and-sleep.  The `notEmpty` condition will be alerted when a session becomes available and this thread will run again to pickup the session.  I don't really understand why there is a timeout currently implemented.

Reply:
Right, this is done as part of putSessionBack() method -- so I dont see an issue here